### DISCUSSION TITLE
Keep old route overlay after rerouting

### DISCUSSION THREAD

**Discussion by mkrezdo** [devemux86/cruiser#554] [Q&A] [answered]
Hi,

today I recognized that after rerouting (because I missed the route) two things will happen:
- after rerouting the "Start" is placed on actual position - OK I never understood this, but it has been ever like that I guess
- after rerouting the Overlay of the "past" route is missing. Ther is only a overlay for the rest of the route then (base on the new Start point). The thing is - sometimes it starts to reroute but I turn around after perhaps 1 km to go back to the old track. But if this is not shown any more (and the navigation is already doing the new route) it's a bit stupid. Then I've to top an load the Overlay of the complete route again.

Am I doing sth wrong or is this how it should be?

Marco

**Comment by devemux86** [comment]
Hi @mkrezdo,

You can _disable the automatic rerouting_, if you do not need it:
- In "Settings | Navigation | Recalculate route"
- Or long press the direction icon at top-left

You can then freely return to your route or press the off-route warning button to make a manual rerouting whenever you like.

About your questions, after a rerouting a new route has been calculated, but only from the user location to the next waypoint.
So the rest of your route is unchanged.

**Reply by bios4** [reply]
> Or long press the direction icon at top-left

Does enabling/disabling rerouting per the direction icon just affect the current (loaded) route, or does this change the profile setting?

**Reply by devemux86** [reply]
Navigation options screen changes the relevant settings, so they are stored in the profile.
Similar to all other options changed through the various panels.
e.g. tap distance panel to switch between destination or selected via point.

**Reply by bios4** [reply]
Thanks for the clarification.

> Navigation options screen changes the relevant settings, so they are stored in the profile.

I wonder if it would make sense to change this behaviour to make these on-screen-changes only affect the settings TEMPORARILY!?
Or are there reasons to not do this?

**Reply by devemux86** [reply]
Imagine we navigate, switch on / off some options, stop somewhere, turn off navigation or Cruiser.
After some time we restart navigation to continue our journey, the options should be as we left them.

Or users will be confused about the different app behavior.

**Reply by bios4** [reply]
That's true...

**Comment by devemux86** [comment]
@mkrezdo your idea to keep the old route overlay after rerouting is interesting.

What do other users think about it?

**Reply by GuG1050** [reply]
I Like this Idea.

**Reply by bios4** [reply]
I also like the idea of keeping the initial/original route overlay intact and visible!

**Reply by DIrgi** [reply]
Me too, at least on demand (activate in setup options).

**Reply by 0709wiwiwi** [reply]
This does not need another extra set up. All necessary stuff is already there.
![image](https://github.com/devemux86/cruiser/assets/54167201/29c3fec3-3af6-403d-8985-cf4da87f1a28)
Requested only is to see the previous display identically after a new restart.

**Reply by mkrezdo** [reply]
I've exactly the same understanding. Keeping the route/overlay ever and if it should be deleted this can already be done.

perhpas one additional idea - but not that important:
If there are diffrent overlays loaded during clearing process a popup could ask if I only want to delete the addition overlays (but keep actual route+overlay) or delete all overlays. Or to make it possible to decide whiche overlays should be deleted (by showing the file names).

**Comment by ramondevic** [comment]
Yes, I like it too

**Comment by devemux86** [comment]
Note that you can already do this if we load the route as overlay.
The route overlay does not change when we recalculate a route.

**Reply by bios4** [reply]
> The route overlay does not change when we recalculate a route.

Obviously this seems to be NOT the case!??

> after rerouting the Overlay of the "past" route is missing. Ther is only a overlay for the rest of the route then (base on the new Start point).

**Reply by mkrezdo** [reply]
is it like that? Finally I loaded it as route + overlay and it got lost after recalcutation of the route. Is there a diffrence if I only load it additionally as overlay? If yes why?
This is also a second question from my side but I could not test it yet. Often I load more variants of a longer route with diffrent colors (I asked for this feature for a longer time for the Kurviger app and I still really like it). What happens with them if I load:
- one route + overlay
- three more routes only as overlay
This is sth I've to test - but also here I'd wish that the overlay will be kept till I delet them via the menu.

**Reply by devemux86** [reply]
> Obviously this seems to be NOT the case!??

@bios4 I meant if we import a route as overlay (thin line).

Not the rendered thick route line, which changes when we calculate a new route.

**Reply by devemux86** [reply]
> I'd wish that the overlay will be kept till I delet them via the menu

The overlays from the favorite routes or tracks always stay until we change their visibility.

If you mean the overlay from the import dialog, it is currently not permanent.
There is a request to remain visible when restarting the app, see here:

- https://github.com/devemux86/cruiser/discussions/431

**Reply by devemux86** [reply]
@mkrezdo 

- If you import a route file, select "Overlay + Routing" in the import dialog.
- If you load a route from the favorites, also set it to be visible as overlay.

In both cases, rerouting only changes the route. The overlay stays unchanged.

**Reply by mkrezdo** [reply]
> @mkrezdo


> * If you load a route from the favorites, also set it to be visible as overlay.
> 
> In both cases, rerouting only changes the route. The overlay stays unchanged.

I'd say that this is not correct. Exactly this was my origin issue. I imported a kurviger file as route+overlay. After rerouting the overlay is get's lost - I only see the overlay of the "new" route that starts at my current position, but I do not see the overlay of the complete and originally imported rout anymore

**Reply by devemux86** [reply]
@mkrezdo I cannot reproduce this, the overlay after rerouting always stays unchanged.

Perhaps you restarted the app after the import?
Currently the overlay from import does not stay after app restart:
- https://github.com/devemux86/cruiser/discussions/431

**Reply by mkrezdo** [reply]
I don't think so. I tested it twice during a tour yesterday. 
Perhaps I can test it again the next day, but could take some time

(normally during driving I do not restart the app :-) )

**Reply by devemux86** [reply]
@mkrezdo not during driving, but after import a route.

Import can be done hours ago and then we close the app.

**Reply by bios4** [reply]
I am now confused to the max... 😕

> @mkrezdo your idea to keep the old route overlay after rerouting is interesting.

This ^^ sounds like you agree that the overlay is removed after rerouting.

> In both cases, rerouting only changes the route. The overlay stays unchanged.

Here you state that the overlay is not changed.

How do these two statements from you fit together @devemux86 ❓

**Reply by devemux86** [reply]
It seems to have confused the 2 lines: thick route line and thin overlay line. For example:
- We import a route with selected the "Overlay + Routing" (image 1)

- We see 2 lines: a thick magenta route line and above a thin blue overlay line (image 2)

- After rerouting the thick magenta route line changes, while the thin blue overlay line remains unchanged (image 3)

The idea suggested in this topic is to draw an overlay line in rerouting for the original route,
when we did not display an overlay like here from the import process.

<img src="https://github.com/devemux86/cruiser/assets/3484020/5cec5fac-dc1d-4323-99b2-2fcd2af3adb8" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/4cf3937a-15c5-4013-a7ef-a94163ae6616" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/5ac55ba9-3f41-4e56-aa69-c0868807b0e9" width="33%"/>

**Reply by mkrezdo** [reply]
> * After rerouting the thick magenta route line changes, while the thin blue overlay line remains unchanged (image 3)

This is not what happend for me. The difference of the lines is clear for me. One is the overlay (thin line) and one the actual route (thick magenta)
I see that there is a difference regarding "what happes after rerouting". I'll try to drive a short distance with the car this evening and then post some screenshots here.

**Reply by bios4** [reply]
> The idea suggested in this topic is to draw an overlay line in rerouting for the original route,
when we did not display an overlay like here from the import process.

My understanding of mkrezdo's initial report is that
- We import a route with selected the "Overlay + Routing" (image 1)
- We see 2 lines: a thick magenta route line and above a thin blue overlay line (image 2)
- After rerouting the thick magenta route line changes, while the thin blue overlay line ~~remains unchanged (image 3)~~ of the initial route is changed/deleted

And the request to change this in a way that the overlay remains unchanged IN ANY CASE, independent from if it was loaded from "My collection", imported from scratch or ...

**Reply by devemux86** [reply]
> After rerouting the thick magenta route line changes, while the thin blue overlay line remains unchanged (image 3) of the initial route is changed/deleted

I cannot reproduce this, @bios4 can you reproduce this?

The imported overlay line remains. Unless restart the app, but this is a different feature request:
- https://github.com/devemux86/cruiser/discussions/431

**Reply by devemux86** [reply]
You don't have to go out to make a test.

Import a route (as overlay + routing) which is a bit far from your location.
Start navigation, select rerouting and see the result on the screen.

**Reply by bios4** [reply]
> I cannot reproduce this, @bios4 can you reproduce this?

> Import a route (as overlay + routing) which is a bit far from your location.
Start navigation, select rerouting and see the result on the screen.

No, at least not by testing with this ^^ method (running 4.1.0-b2)...
Wich app version do you have in use @mkrezdo?

**Reply by mkrezdo** [reply]
I use 4.0.2 - give me some time. I'll test it

**Reply by mkrezdo** [reply, marked as answer]
OK I tested it two times in our village now. The overlay did not get lost. If it is like that it'd be fine. But I'm very sure that it was different during my ride yesterday. But no idea... Perhaps there was an other issue... Sorry for confusing.

But anyway I'd support this other request that overlays should be kept also after closing the app. I often prepare everything in the evening, than it's good to have it still on the next morning.

**Reply by devemux86** [reply]
@mkrezdo thanks for the extra tests!

Probably exited the app, as currently the imported overlays are not saved on exit.
It is already in my list to keep them. :slightly_smiling_face:
